src/main/webapp/app/quiz/edit/drag-and-drop-question/edit-drag-and-drop-question.component.ts

```diff
@@ -839,5 +804,65 @@ export class EditDragAndDropQuestionComponent implements OnInit, OnChanges, Afte
      */
     togglePreview(): void {
         this.showPreview = !this.showPreview;
+        this.prepareForSave();
+    }
+
+    /**
+     * @function changesInMarkdown
+     * @desc Detect of text changes in the markdown editor
+     *      1. Notify the parent component to check the validity of the text
+     *      2. Parse the text in the editor
+     */
+    changesInMarkdown(): void {
+        this.questionUpdated.emit();
+        this.changeDetector.detectChanges();
+        this.prepareForSave();
+    }
+
+    /**
+     * @function domainCommandsFound
+     * @desc Get the {array} from the editor and assign its values based on the domainCommands
+     *       to the corresponding question attributes one by one
+     *       1. Go trough each element of the {array} domain command
+     *       2. If the first element {string} has not second element {domainCommandIdentifier} in the array assign the string to the question text
+     *       3. else assign the first element {string} according to the second element {domainCommandIdentifier} to the corresponding attributes of the question
+     * @param {array} contains markdownTextLine with the corresponding domainCommand {DomainCommand} identifier
+     */
+    domainCommandsFound(domainCommands: [string, DomainCommand][]): void {
+        this.cleanupQuestion();
+
+        for (const command of domainCommands)  {
```

Comment:
It would be easier to understand if you directly destructured the string and command in the for loop and gave them speaking names.
for (const [name, command] of docmainCommands) {...

Reply:
Please rename name into something more self explanatory. It was only meant as a placeholder.